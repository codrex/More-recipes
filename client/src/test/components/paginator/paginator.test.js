/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import toJson from 'enzyme-to-json';
import Paginator from '../../../components/common/Paginator';

const props = {
  pageCount: 5,
  handlePageClick: jest.fn(),
  loading: false
};

describe('Paginator component ', () => {
  test('should render a pagination component', () => {
    const component = shallow(<Paginator {...props} />);
    const tree = toJson(component);
    expect(component.find('nav').length).toBe(1);
    expect(tree).toMatchSnapshot();
  });
});
