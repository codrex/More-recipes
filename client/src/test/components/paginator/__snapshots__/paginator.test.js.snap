// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator component  render as expected when component is mounted 1`] = `
<nav
  aria-label="Page navigation"
  className="d-flex justify-content-center col-12 "
>
  <PaginationBoxView
    activeClassName="page-active"
    breakClassName="break-me"
    breakLabel={
      <a
        href=""
      >
        ...
      </a>
    }
    containerClassName="pagination justify-content-center"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={2}
    nextClassName="page-item page-link"
    nextLabel={
      <Icon
        active=""
        className=""
        dataTarget=""
        dataToggle=""
        handleClick={[Function]}
        iconClass="fa fa-angle-double-right"
        onMouseEnter={[Function]}
        parentClass=""
        pointer={false}
        tip="hello"
        toolTip={false}
        toolTipId=""
        toolTipPlace="bottom"
        toolTipType="dark"
      />
    }
    onPageChange={[Function]}
    pageClassName="page-item"
    pageCount={5}
    pageLinkClassName="page-link"
    pageRangeDisplayed={5}
    previousClassName="page-item page-link"
    previousLabel={
      <Icon
        active=""
        className=""
        dataTarget=""
        dataToggle=""
        handleClick={[Function]}
        iconClass="fa fa-angle-double-left"
        onMouseEnter={[Function]}
        parentClass=""
        pointer={false}
        tip="hello"
        toolTip={false}
        toolTipId=""
        toolTipPlace="bottom"
        toolTipType="dark"
      />
    }
    subContainerClassName="page-item"
  />
</nav>
`;
